asv/util.py

```diff
@@ -346,75 +354,85 @@ def get_content(header=None):
     stdout_chunks = []
     stderr_chunks = []
     is_timeout = False
-    try:
-        if posix:
-            # Forward signals related to Ctrl-Z handling; the child
-            # process is in a separate process group so it won't receive
-            # these automatically from the terminal
-            def sig_forward(signum, frame):
-                os.killpg(proc.pid, signum)
-                if signum == signal.SIGTSTP:
-                    os.kill(os.getpid(), signal.SIGSTOP)
-            signal.signal(signal.SIGTSTP, sig_forward)
-            signal.signal(signal.SIGCONT, sig_forward)
-
-        fds = {
-            proc.stdout.fileno(): stdout_chunks,
-            proc.stderr.fileno(): stderr_chunks
-            }
-
-        while proc.poll() is None:
-            try:
-                rlist, wlist, xlist = select.select(
-                    list(fds.keys()), [], [], timeout)
-            except select.error as err:
-                if err.args[0] == errno.EINTR:
-                    # interrupted by signal handler; try again
-                    continue
-
-            if len(rlist) == 0:
-                # We got a timeout
-                is_timeout = True
-                break
-            for f in rlist:
-                output = os.read(f, PIPE_BUF)
-                fds[f].append(output)
-            if dots and time.time() - last_dot_time > 0.5:
-                if dots is True:
-                    log.dot()
-                elif dots:
-                    dots()
-                last_dot_time = time.time()
-    finally:
-        if posix:
-            # Restore signal handlers
-            signal.signal(signal.SIGTSTP, signal.SIG_DFL)
-            signal.signal(signal.SIGCONT, signal.SIG_DFL)
-
-        if proc.returncode is None:
-            # Timeout or another exceptional condition occurred, and
-            # the program is still running.
+    if WIN:
+        try:
+            stdout, stderr = proc.communicate(timeout=timeout)
+        except subprocess.TimeoutExpired:
+            proc.kill()
+            is_timeout = True
+    else:
+        try:
             if posix:
```

Comment:
Wrong indentation (doesn't do what the original code did) -- only the signal.signal calls should be inside `if posix`